impls/gsnforwarder/scripts/generate-bindings.sh

```diff
@@ -1,36 +1,41 @@
 #!/bin/bash
 
-# Use the project root (one directory above this script) as the current working directory:
-PROJECT_ROOT="$(dirname "$(dirname "$(readlink -f "$0")")")"
-cd "${PROJECT_ROOT}" || exit 1
+# Change to the gsnforwarder dir, one directory up from this script
+GSNFORWARDER_DIR="$(dirname "$(dirname "$(readlink -f "$0")")")"
+cd "${GSNFORWARDER_DIR}" || exit 1
 
 ABIGEN="$(go env GOPATH)/bin/abigen"
 
 PKG_NAME="gsnforwarder"
 
-if [[ -z "${SOLC_BIN}" ]]; then
-	SOLC_BIN=solc
-fi
+rm -rf contracts/abi contracts/bin
 
-"${SOLC_BIN}" --abi contracts/Forwarder.sol -o contracts/abi/ --overwrite
-"${SOLC_BIN}" --bin contracts/Forwarder.sol -o contracts/bin/ --overwrite
+# We need the solc version to match the contract deployed here:
+# https://docs.opengsn.org/networks/ethereum/mainnet.html
+# https://etherscan.io/address/0xB2b5841DBeF766d4b521221732F9B618fCf34A87#code
+SOLC_VERSION="0.8.7"
+
+SOLC_EXEC=(
+	docker run --rm
+	--volume "${GSNFORWARDER_DIR}/contracts:/contracts"
+	--workdir /contracts
+	-u "$(id -u):$(id -g)"
+	"ethereum/solc:${SOLC_VERSION}"
+)
+
+SOLC_EXEC+=(--allow-paths . --optimize)
+
+set -x
+"${SOLC_EXEC[@]}" --abi forwarder/Forwarder.sol -o abi/ --overwrite
+"${SOLC_EXEC[@]}" --bin forwarder/Forwarder.sol -o bin/ --overwrite
```

Comment:
For using a very specific, older version of solc, using docker seems like the best way to go.